dockertest/subtestbase.py

```diff
@@ -155,8 +157,15 @@ def failif_not_in(needle, haystack, description=None):
             description = 'string'
         if needle in haystack:
             return
-        raise DockerTestFail("Expected string '%s' not in %s '%s'"
-                             % (needle, description, haystack))
+        expect_s = "Expected string '%s'" % needle
+        if '|' in needle:
+            needles = [n.strip() for n in needle.split('|')]
+            expect_s = "Expected strings %s" % needles
+            for subneedle in needles:
+                if subneedle.strip() in haystack:
+                    return
+        raise DockerTestFail("%s not in %s '%s'" %
+                             (expect_s, description, haystack))
```

Comment:
>Can we say with confidence that the '|' character will never be needed as part of the search string?

No, but I'm reluctant to add complexity for what seems like a low-probability risk. If this code becomes wildly popular, with millions of users worldwide, someone will inevitably find a case where they need '|', and it will fail, and they will curse me. For our small limited-case purposes, a handful of developers on a tiny scale, I can't justify the code complexity.